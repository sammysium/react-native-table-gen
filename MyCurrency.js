import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { formatCurrency }  from './helperFunctions';

const MyComponent = ({
  amount,
  currencySymbol,
  curencySymbolPosition,
  decimalsToShow,
  styles,
  rest
}) => {
  let txtStyle = styles !== undefined ? styles : defaultStyles.text;
  const output = formatCurrency(
    amount,
    currencySymbol,
    curencySymbolPosition,
    decimalsToShow
  );

  return (
    <>
      <Text style={txtStyle} {...rest}>{output}</Text>
    </>
  );
};

const defaultStyles = StyleSheet.create({
  text: {
    color: '#000',
    marginBottom: 4,
    marginRight: 10
  }
});
MyComponent.propTypes = {
  amount: PropTypes.number.isRequired,
  decimalsToShow: PropTypes.number,
  curencySymbolPosition: PropTypes.string,
  styles: PropTypes.instanceOf(Object),
  currencySymbol : PropTypes.string.isRequired,
  rest: PropTypes.instanceOf(Object)
};
MyComponent.defaultProps ={
  styles: undefined,
  decimalsToShow: 2,
  curencySymbolPosition: 'front',
  rest : {}
};
const MyCurrency = React.memo(MyComponent);
export default MyCurrency;
