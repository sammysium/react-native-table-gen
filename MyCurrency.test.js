import React from 'react';
import { Text } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';
import MyCurrency from './MyCurrency';

describe('component MyCurrency', () => {
  const amount = 2989;
  const currency = "rwf"
  it('renders correctly', () => {
    const snapShotTree = render(<MyCurrency amount={amount} currencySymbol={currency} />).toJSON();
    expect(snapShotTree).toMatchSnapshot();
  });

  it('should return formatted currency', () => {
    const App = () => (
      <>
        <MyCurrency amount={amount} currencySymbol={currency} />
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `${currency} 2,989.00`;
    expect(label.props.children).toStrictEqual(lbl);
  });

  it('should return formatted currency - symbol back', () => {
    const App = () => (
      <>
        <MyCurrency amount={amount} curencySymbolPosition="back"  currencySymbol={currency}/>
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `2,989.00 ${currency}`;
    expect(label.props.children).toStrictEqual(lbl);
  });

  it('should be pressed ', () => {
    const events = {
      onPress: jest.fn(()=>{
        console.log('pressed');
      })
    }
    const App = () => (
      <>
        <MyCurrency amount={amount} {...events} currencySymbol={currency}/>
      </>
    );
      const spy = jest.spyOn(console,'log')
    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    fireEvent.press(label)
    expect(spy).toHaveBeenCalled();
  });
});
