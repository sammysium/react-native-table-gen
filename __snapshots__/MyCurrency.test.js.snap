// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component MyCurrency renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#000",
      "marginBottom": 4,
      "marginRight": 10,
    }
  }
>
  rwf 2,989.00
</Text>
`;
